cmd/connect.go

```diff
@@ -55,7 +55,17 @@ type ConnectOptions struct {
 
 // Complete adds any missing configuration that can only be added just before calling Run
 func (o *ConnectOptions) Complete(f Factory, args []string) (err error) {
-	if o.Setup {
+	repoPath := f.RepoPath()
+
+	repoErr := lib.QriRepoExists(repoPath)
+
+	// If we have a repo, we should not acknowledge the `setup` flag. This is
+	// specifically important for any automated running of the qri backend that
+	// cannot know before hand if the user has a repository setup or not.
+	// The `--setup` flag, in this case, is intended as a catch-all: if the user
+	// has not set up a repository yet, run the setup before running `qri connect`,
+	// otherwise, continue to spin up the daemon.
```

Comment:
Went back and forth with the content of this comment. Would love feedback.

Reply:
First, don't think we need the comment. Instead let's focus on making the logic clear.
I think this should be inverted, where we check for repo existence only if `o.Setup` is true, and the solution should be ever-so-slightly more thorough:

```go
if o.Setup {
  repoErr := lib.QriRepoExists(f.RepoPath())
  if errors.Is(repoErr, lib.ErrNoRepo) {
     // make a repo...
  } else if repoErr != nil {
    return err
  }
```

this'll require modifying `lib/setup.go`:
```go
var ErrNoRepo = fmt.Errorf("no repo exists")

// QriRepoExists returns nil if a qri repo is defined at the given path
// does not attempt to open the repo
func QriRepoExists(path string) error {
	// for now this just checks for an existing config file
	_, err := os.Stat(filepath.Join(path, "config.yaml"))
	if !os.IsNotExist(err) {
		return nil
	}
	return ErrNoRepo
}
```

note the return value is no longer `repo.ErrNoRepo`. This might break a test or two, but really shouldn't. We're declaring the `ErrNoRepo` in lib to avoid having to import repo within `cmd`.